File: truth_set/go/logic/wrong_location/configManager.go
package wrong_location

import (
    "fmt"
    "sync"
    "time"
)

type ConfigManager struct {
    config map[string]string
}

var instance *ConfigManager
var once sync.Once

func GetInstance() *ConfigManager {
    once.Do(func() {
        instance = &ConfigManager{}
        go instance.loadConfig() 
    })
    return instance
}

func (cm *ConfigManager) loadConfig() {
    time.Sleep(1 * time.Second)
    cm.config = map[string]string{
        "apiUrl": "https://api.example.com",
    }
    fmt.Println("Config loaded:", cm.config)
}

func main() {
    config := GetInstance()
    fmt.Println("API URL:", config.config["apiUrl"])
}
\ No newline at end of file

File: src/typescript/logic/wrong_location/configManager.ts
class ConfigManager {
    private static instance: ConfigManager;
    config: any;

    private constructor() {}

    static getInstance(): ConfigManager {
        if (!ConfigManager.instance) {
            ConfigManager.instance = new ConfigManager();
            ConfigManager.instance.loadConfig();
        }
        return ConfigManager.instance;
    }

    async loadConfig() {
        this.config = await new Promise(resolve => setTimeout(() => resolve({ apiUrl: 'https://api.example.com' }), 1000));
        console.log('Config loaded:', this.config);
    }
}

const config = ConfigManager.getInstance();
console.log('API URL:', config.config.apiUrl);
\ No newline at end of file